### PR TITLE
Removed HeaderCarrier, changed param names, added Alert levels

### PR DESCRIPTION
**Pull request by scottcutts**

### CODE REVIEW

File: src/main/scala/uk/gov/hmrc/play/events/Recordable.scala [hunk 1]
```diff
@@ -26,15 +26,14 @@ trait Auditable extends Recordable {
   def source: String
   def name: String
   def tags: Map[String, String]
-  def details: Map[String, String]
-  def headerCarrier: HeaderCarrier
```
Comment (liquidarmour) [inline src/main/scala/uk/gov/hmrc/play/events/Recordable.scala]: Yay

File: src/main/scala/uk/gov/hmrc/play/events/Recordable.scala [hunk 2]
```diff
@@ -45,14 +44,14 @@ trait Measurable extends Recordable {
...
-  def details: Map[String, String]
+  def data: Map[String, String]
 
 }
 
 object AlertLevel extends Enumeration {
 
   type AlertLevel = Value
-  val WARNING, CRITICAL = Value
+  val WARNING, MINOR, MAJOR, CRITICAL = Value
```
Comment (liquidarmour) [inline src/main/scala/uk/gov/hmrc/play/events/Recordable.scala]: Makes sense to me.

File: src/main/scala/uk/gov/hmrc/play/events/AuditEventHandler.scala [hunk 3]
```diff
@@ -17,11 +17,12 @@
 package uk.gov.hmrc.play.events
 
 import play.api.libs.concurrent.Execution.Implicits.defaultContext
+import uk.gov.hmrc.play.audit.http.HeaderCarrier
 import uk.gov.hmrc.play.audit.http.connector.AuditConnector
 
 import scala.concurrent.Future
 
-trait AuditEventHandler extends EventHandler {
+abstract class AuditEventHandler(implicit headerCarrier: HeaderCarrier) extends EventHandler {
```
Comment (liquidarmour) [inline src/main/scala/uk/gov/hmrc/play/events/AuditEventHandler.scala]: Actually, depending how this is used will it always have the current requests headerCarrier?

Comment (scottcutts) [inline src/main/scala/uk/gov/hmrc/play/events/AuditEventHandler.scala]: Yes, the auditConnector interface used here always requires a HeaderCarrier. If you changed it, you'd have to write your own AuditEventHandler anyway.

**Pull request merged**